### PR TITLE
CXP-1187: Add catalog scopes

### PR DESCRIPTION
**Pull request by tseho**
**Description (for Contributor and Core Developer)**

It was requested that the ACLs behind the new scopes are not visible from the UI, to do so, we added a way of hiding ACLs in this commit https://github.com/akeneo/pim-community-dev/pull/16963/commits/9c8e16c1aaebe91b6024f769b85b577480c44d65

Screenshot_2022-05-06_18-29-01

Screenshot_2022-05-06_18-28-35

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

**Review by guillaumefer-akeneo — APPROVED** [review]

**Comment by ahocquard** [comment]
Why not the same solution as for feature flags?

**Comment by tseho** [comment]
> Why not the same solution as for feature flags?
Because I want the ACL to still exist.

**Review by RiffVentura — APPROVED** [review]

File: components/catalogs/back/tests/Unit/Infrastructure/Security/CatalogScopeMapperTest.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                'scope' => 'unknown',
+                'result' => [],
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider messages
+     */
+    public function testItReturnsMessagesForOneScope(string $scope, $expected): void
```
Comment (masacc) [inline components/catalogs/back/tests/Unit/Infrastructure/Security/CatalogScopeMapperTest.php:65]: I wonder if that kind of test is really useful. I mean it's almost like testing constant and array access.

**Review by masacc — APPROVED** [review]

**Review by willy-ahva — APPROVED** [review]

File: components/catalogs/back/tests/Unit/Infrastructure/Security/CatalogScopeMapperTest.php (hunk 1 above)
Comment (tseho) [reply components/catalogs/back/tests/Unit/Infrastructure/Security/CatalogScopeMapperTest.php:65]: Agree, but the ScopeRegistry will call the methods not the constants, it's only there to guarantee that the internal logic is handled correctly.

It's not the most valuable test but it's a low cost one.

**Pull request merged**